llvm/lib/Target/AMDGPU/AMDGPUGlobalISelUtils.cpp

```diff
@@ -106,3 +106,59 @@ void IntrinsicLaneMaskAnalyzer::findLCSSAPhi(Register Reg) {
       S32S64LaneMask.insert(LCSSAPhi.getOperand(0).getReg());
   }
 }
+
+static LLT getReadAnyLaneSplitTy(LLT Ty) {
+  if (Ty.isVector()) {
+    LLT ElTy = Ty.getElementType();
+    if (ElTy == LLT::scalar(16))
+      return LLT::fixed_vector(2, 16);
```

Comment:
```suggestion
    if (ElTy.getSizeInBits() == 16)
      return LLT::fixed_vector(2, EltTy);
```
Future proofing for float LLT